lib/delegate/proxy.js

```diff
@@ -137,29 +138,41 @@ var proto = {
         response.once('error', this.onComplete);
         response.once('finish', this.onComplete);
 
-        Wreck.read(res, { json: true }, function (err, payload) {
-            var response;
-
-            if (err) {
-                // Actual parsing error, so fail immediately.
-                // The server returned an invalid response
-                // so the client should be notified.
-                err = Hapi.error.wrap(err);
-                err.output.headers['x-registry'] = self.registry;
-                reply(err);
-                return;
-            }
-
-            response = reply(payload).code(res.statusCode);
-            response.header('x-registry', self.registry);
-
-            Object.keys(res.headers)
-                .filter(Util.isValidHeader)
-                .forEach(function (key) {
-                    response.header(key, res.headers[key]);
-                });
-        });
+        contentType = res.headers['content-type'];
+        contentType = contentType && ContentType.parse(contentType).type; // lol
```

Comment:
Yup. Also, as for why the string instead of req/res: was unable to parse anything but the string in the tests. Not sure if it's because of proxy or because of our :monkey:ing with the request.